### DISCUSSION TITLE
Implement multiple loop variables for for loops

### DISCUSSION THREAD

**Discussion by InkONat** [godotengine/godot-proposals#8689] [Scripting]
Implement multiple loop variables for for loops:

```
var array1 = ["apple", "banana"]
var array2 = ["John", "Mario"]

for favorite_fruit, name in array1, array2:
    print(name, "s favorite fruit is ", favorite_fruit) # Prints "Johns favorite fruit is apple", "Marios favorite fruit is banana"

```

I know that the example I used could be done using a dictionary, but I think especially for more loop variables there could be many use cases for multiple loop variables.

Also, for 2 loop variables, you should also be able to iterate over a dictionary, with the first loop variable being the key and the second the value. The workaround is, of course, just iterating over the keys and for the values do `dictionary[key]`, but I think you should still be able to use this 2 loop variable alternative.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/3457.
